code/game/objects/items/weapons/circuitboards/machinery/chemistry.dm

```diff
@@ -16,3 +16,46 @@
 /obj/item/stock_parts/circuitboard/reagent_heater/cooler
 	name = "circuitboard (chemical cooler)"
 	build_path = /obj/machinery/reagent_temperature/cooler
+
+/obj/item/stock_parts/circuitboard/chem_master
+	name = "circuitboard (ChemMaster 3000)"
+	build_path = /obj/machinery/chem_master
+	board_type = "machine"
+	origin_tech = "{'biotech':2,'engineering':1}"
+	req_components = list(
+		/obj/item/stock_parts/matter_bin = 1,
+		/obj/item/stock_parts/manipulator = 1,
+		/obj/item/stock_parts/scanning_module = 1
+	)
+	additional_spawn_components = list(
+		/obj/item/stock_parts/console_screen = 1,
+		/obj/item/stock_parts/keyboard = 1,
+		/obj/item/stock_parts/power/apc/buildable = 1
+	)
+
+/obj/item/stock_parts/circuitboard/chemical_dispenser
+	name = "circuitboard (chemical dispenser)"
+	build_path = /obj/machinery/chemical_dispenser
+	board_type = "machine"
+	origin_tech = "{'biotech':2,'engineering':1}"
+	req_components = list(
+		/obj/item/stock_parts/matter_bin = 2,
+		/obj/item/stock_parts/manipulator = 1
+	)
+	additional_spawn_components = list(
+		/obj/item/stock_parts/console_screen = 1,
+		/obj/item/stock_parts/keyboard = 1,
+		/obj/item/stock_parts/power/apc/buildable = 1
+	)
+
+/obj/item/stock_parts/circuitboard/chemical_dispenser/bar_alc
+	name = "circuitboard (chemical dispenser)"
+	build_path = /obj/machinery/chemical_dispenser/bar_alc
+
+/obj/item/stock_parts/circuitboard/chemical_dispenser/bar_soft
+	name = "circuitboard (chemical dispenser)"
+	build_path = /obj/machinery/chemical_dispenser/bar_soft
+
+/obj/item/stock_parts/circuitboard/chemical_dispenser/bar_coffee
+	name = "circuitboard (chemical dispenser)"
+	build_path = /obj/machinery/chemical_dispenser/bar_coffee
```

Comment:
I suggest using the subtypes-via-multitool thing on some of these, to lower circuit type bloat.